ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -685,6 +685,40 @@ private synchronized void sendStartLeaderElectionToHigherPriorityPeer(RaftPeerId
     });
   }
 
+  boolean sendStartLeaderElection(FollowerInfo followerInfo, int leaderPriority) {
```

Comment:
This abstraction is quite hard to be reused in TransferLeadership.
Because TransferLeadershipRequest contains `RaftPeerId transferee` instead of `FollowerInfo`. And `leaderPriority` sounds like an internal state of `LeaderStateImpl`.

Also, I think it's probably better to use another name instead of overload `sendStartLeaderElection`, because it's doing something else. And it may not send StartLeaderElection if conditions are not met.

Reply:
We will get `FollowerInfo` (it is a must since we need match index) and then pass null for `leaderPriority`.

Reply:
> and then pass null for `leaderPriority`.

`leaderPriority` is int, we cannot pass null to it.

And I think we should also check priority in TransferLeadership, otherwise `checkPeersForYieldingLeader()` will transfer it back very soon.

Reply:
> ... find the peer with max priority first ...

After the change above, we no longer need to pass `leaderPriority` here.